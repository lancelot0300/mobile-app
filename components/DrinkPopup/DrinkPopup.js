import React, { useState } from "react";
import {
  Alert,
  Modal,
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Image,
  SafeAreaView,
} from "react-native";
import closeIcon from "../../assets/close-red-icon.png";
import drinkIcon from "../../assets/drink.png";


const DrinkPopup = ({ setModalVisible, modalVisible, drink }) => {
  return (
    <>
      {modalVisible && drink && (
        <SafeAreaView style={styles.centeredView}>
          <Modal animationType="fade" visible={modalVisible} transparent={true}>
            <SafeAreaView style={{ backgroundColor: "#121212" }}>
              <TouchableOpacity
                onPress={() => {
                  setModalVisible(!modalVisible);
                }}
                style={{
                  backgroundColor: "#fff",
                  width: 40,
                  height: 40,
                  borderRadius: 20,
                  alignSelf: "flex-end",
                  marginTop: 10,
                  marginRight: 10,
                }}
              >
                <Image style={styles.closeIcon} source={closeIcon} />
              </TouchableOpacity>
            </SafeAreaView>
            <View style={styles.centeredView}>
              <Text style={styles.nameText}>{drink.name}</Text>
              <Image
                source={drinkIcon}
                style={{
                  width: 150,
                  height: 150,
                  alignSelf: "center",
                  marginBottom: 10,
                }}
              />
              <Text style={styles.headerText}>Ingredients:</Text>
              {drink.ingredients.map((ingredient, i) => (
                <Text key={i} style={styles.modalText}>
                  - {ingredient}
                </Text>
              ))}
              <Text style={styles.headerText}>Instructions:</Text>
              <Text style={styles.modalText}>{drink.instructions}</Text>
            </View>
          </Modal>
        </SafeAreaView>
      )}
    </>
  );
};

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#121212",
  },
  modalView: {
    margin: 20,
    backgroundColor: "#121212",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#fff",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,

    elevation: 5,
  },
  modalText: {
    marginBottom: 15,
    color: "#fff",
    textAlign: "center",
  },
  headerText: {
    fontSize: 24,
    marginBottom: 15,
    color: "#fff",
    textAlign: "center",
  },
  nameText: {
    fontSize: 30,
    marginBottom: 15,
    color: "#fff",
    textAlign: "center",
  },
  closeIcon: {
    width: 40,
    height: 40,
  },
});

export default DrinkPopup;
