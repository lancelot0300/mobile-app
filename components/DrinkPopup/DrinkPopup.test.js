import React from 'react';
import renderer from 'react-test-renderer';

import  DrinkPopup from './DrinkPopup';


describe('<DrinkPopup />', () => {

    it('renders correctly', () => {
        const tree = renderer.create(<DrinkPopup
            setModalVisible={true}
            modalVisible={true}
            drink={{
                "name": "drink1",
                "ingredients": ["ingedient1", "ingedient2", "ingedient3"],
                "instructions": "instructions"
            }}
            />).toJSON();
        expect(tree).toMatchSnapshot();
    }
    );

    it('not showing popup without prop modalVisible', () => {
        const tree = renderer.create(<DrinkPopup  setModalVisible={true}
            drink={{
                "name": "drink1",
                "ingredients": ["ingedient1", "ingedient2", "ingedient3"],
                "instructions": "instructions"
            }} />).toJSON();
        expect(tree).toBe(null);
    }
    );

    it('not showing popup without props', () => {
        const tree = renderer.create(<DrinkPopup  />).toJSON();
        expect(tree).toBe(null);
    }
    );

});