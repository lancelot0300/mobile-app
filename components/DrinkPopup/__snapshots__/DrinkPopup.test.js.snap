// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrinkPopup /> renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#121212",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <RCTSafeAreaView
      style={
        {
          "backgroundColor": "#121212",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-end",
            "backgroundColor": "#fff",
            "borderRadius": 20,
            "height": 40,
            "marginRight": 10,
            "marginTop": 10,
            "opacity": 1,
            "width": 40,
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
    </RCTSafeAreaView>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#121212",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 30,
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        drink1
      </Text>
      <Image
        source={1}
        style={
          {
            "alignSelf": "center",
            "height": 150,
            "marginBottom": 10,
            "width": 150,
          }
        }
      />
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 24,
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        Ingredients:
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        - 
        ingedient1
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        - 
        ingedient2
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        - 
        ingedient3
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 24,
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        Instructions:
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        instructions
      </Text>
    </View>
  </Modal>
</RCTSafeAreaView>
`;
